🩹 Fix limiter middleware db connection

qracer opened the pull request:
Closes #1812

This PR just removes `release` method from [manager.go](https://github.com/gofiber/fiber/blob/master/middleware/limiter/manager.go) so `SkipSuccessfulRequests` and `SkipFailedRequests` from the limiter config will work properly.

Close my PR if you find my approach inappropriate to merge. I just used the simplest way to fix the bug but it can be fixed via refactoring code in [limited_config.go](https://github.com/gofiber/fiber/blob/master/middleware/limiter/limiter_fixed.go)

ReneWerner87 commented:
i will check this tomorrow

ReneWerner87 commented:
@qracer I think releasing the memory and putting it back into the pool is important for the performance.

You are right, if another storage is used, the following construct makes no sense

I think we should execute the get on the key  in the body of the block and set it again after the change, then the releasing can also remain in it.

Should work or?

qracer commented:
@ReneWerner87 I reckon we can do without calling `get` twice. A more convenient solution would be to rearrange the mutex schedule and postpone `set` execution with `defer`. I will commit a new bugfix this evening.

ReneWerner87 commented:
Ok.

Thought one the position is binding, because one has only after the next the results.
I'm curious how your fix looks like.
Best would also be more advanced tests that test just that.

qracer commented:
@ReneWerner87 I've updated the pull request and changed the bugfix approach. Sorry for no test, I don't know  how to mock every available database efficiently to test

ReneWerner87 commented:
Ok, no problem

ReneWerner87 commented on middleware/limiter/limiter_fixed.go:
```diff
@@ -69,12 +72,6 @@ func (FixedWindow) New(cfg Config) fiber.Handler {
 		remaining := cfg.Max - e.currHits
 
-		// Update storage
-		manager.set(key, e, cfg.Expiration)
```
think moving the set function must also be done in the part where the limit was reached

ReneWerner87 commented on middleware/limiter/limiter_fixed.go:
```diff
@@ -44,6 +44,9 @@ func (FixedWindow) New(cfg Config) fiber.Handler {
 		// Lock entry
 		mux.Lock()
+		defer func() {
+			mux.Unlock()
```
this is not a good idea, the mutex in the defer would block the reading for too long

Because with c.Next() all possible handlers are executed

ReneWerner87 commented:
@qracer i added some comments, can you please check it again
